pkg/operation/botanist/component/dependencywatchdog/bootstrap.go

```diff
@@ -452,27 +416,134 @@ func (b *bootstrapper) podLabels() map[string]string {
+func (b *bootstrapper) getContainerCommand() []string {
 	switch b.values.Role {
-	case RoleEndpoint:
+	case RoleWeeder:
 		return []string{
 			"/usr/local/bin/dependency-watchdog",
+			"weeder",
 			fmt.Sprintf("--config-file=%s/%s", volumeMountPath, configFileName),
-			"--deployed-namespace=" + b.namespace,
-			"--watch-duration=5m",
+			"--enable-leader-election=true",
 		}
 
-	case RoleProbe:
+	case RoleProber:
 		return []string{
 			"/usr/local/bin/dependency-watchdog",
-			"probe",
+			"prober",
 			fmt.Sprintf("--config-file=%s/%s", volumeMountPath, configFileName),
-			"--deployed-namespace=" + b.namespace,
-			"--qps=20.0",
-			"--burst=100",
-			"--v=4",
+			"--kube-api-qps=20.0",
+			"--kube-api-burst=100",
+			"--zap-log-level=INFO",
+			"--enable-leader-election=true",
 		}
 	}
 
 	return nil
 }
+
+func (b *bootstrapper) getDeployment(serviceAccName string, configMapName string) *appsv1.Deployment {
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      b.name(),
+			Namespace: b.namespace,
+			Labels: utils.MergeStringMaps(map[string]string{
+				resourcesv1alpha1.HighAvailabilityConfigType: resourcesv1alpha1.HighAvailabilityConfigTypeController,
+			}, b.getLabels()),
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas:             pointer.Int32(1),
+			RevisionHistoryLimit: pointer.Int32(2),
+			Selector:             &metav1.LabelSelector{MatchLabels: b.getLabels()},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: b.getPodLabels(),
+				},
+				Spec: corev1.PodSpec{
+					PriorityClassName:             v1beta1constants.PriorityClassNameSeedSystem800,
+					ServiceAccountName:            serviceAccName,
+					TerminationGracePeriodSeconds: pointer.Int64(5),
+					Containers: []corev1.Container{{
+						Name:            prefixDependencyWatchdog,
+						Image:           b.values.Image,
+						ImagePullPolicy: corev1.PullIfNotPresent,
+						Command:         b.getContainerCommand(),
+						Ports: []corev1.ContainerPort{{
+							Name:          "metrics",
+							ContainerPort: 9643,
+							Protocol:      corev1.ProtocolTCP,
+						}},
+						Resources: corev1.ResourceRequirements{
+							Requests: corev1.ResourceList{
+								corev1.ResourceCPU:    resource.MustParse("200m"),
+								corev1.ResourceMemory: resource.MustParse("256Mi"),
+							},
+							Limits: corev1.ResourceList{
+								corev1.ResourceMemory: resource.MustParse("512Mi"),
+							},
+						},
+						VolumeMounts: []corev1.VolumeMount{{
+							Name:      volumeName,
+							MountPath: volumeMountPath,
+							ReadOnly:  true,
+						}},
+					}},
+					Volumes: []corev1.Volume{{
+						Name: volumeName,
+						VolumeSource: corev1.VolumeSource{
+							ConfigMap: &corev1.ConfigMapVolumeSource{
+								LocalObjectReference: corev1.LocalObjectReference{
+									Name: configMapName,
+								},
+							},
+						},
+					}},
+				},
+			},
+		},
+	}
```

Comment:
If you now have a dedicated function for the deployment, then you should call `InjectAnnotations` here.